source/extensions/common/aws/credentials_provider_impl.cc

```diff
@@ -435,14 +436,21 @@ TaskRoleCredentialsProvider::TaskRoleCredentialsProvider(
           envoy::config::cluster::v3::Cluster::STATIC /*cluster_type*/, credential_uri),
       credential_uri_(credential_uri), authorization_token_(authorization_token) {}
 
-bool TaskRoleCredentialsProvider::needsRefresh() {
+bool ContainerRoleCredentialsProvider::needsRefresh() {
   const auto now = api_.timeSource().systemTime();
   return (now - last_updated_ > REFRESH_INTERVAL) ||
          (expiration_time_ - now < REFRESH_GRACE_PERIOD);
 }
 
-void TaskRoleCredentialsProvider::refresh() {
-  ENVOY_LOG(debug, "Getting AWS credentials from the task role at URI: {}", credential_uri_);
+void ContainerRoleCredentialsProvider::refresh() {
+  ENVOY_LOG(debug, "Getting AWS credentials from the container role at URI: {}", credential_uri_);
+
+  // EKS Pod Identity token is sourced from AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE
+  auto token = Utility::getAuthorizationTokenFromEnvFile(AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE);
```

Comment:
This is the key part of this change - auth token will take the contents of the token file if path is provided via this environment variable